### ISSUE TITLE
issue with demultiplexing 

### ISSUE THREAD

**Issue by mehdiamiri123** [marcelm/cutadapt#594]
Cutadapt 3.4 
Python 3.9.6
PiP
Command line parameters: -j 12 -a AGATCGGAAGAGCACACGTCTGAA -o filename_clipped.fastq.gz  filename.fastq.gz
Command line parameters: -e 0.2 --minimum-length=6 --nextseq-trim=28 --no-indels -a NI813=CTAGA -a NI814=GATCA -a NI815=GCATA -a NI816=TAGAC -a NI817=TCTAG -o filename_{name}.fastq.gz filename_clipped.fastq.gz

I am trying to demultiplex a file which contains five different barcode (NI813-NI817), I remove the constant adapter with the fitst command which produces the expected output, one example below

@A00564:455:H3HKYDSX3:3:1101:1380:1000 1:N:0:ACTTGA
NTATCGCCCACGTGGACCATGGCAAGTCCACGCTGACGTTGCTAGAC**AGATCGGAAGAGCACACGTCTGAA**CTCCAGTCACACTTGAATCTCGTATTCCGTCTTCTGCTTGAAAAAGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG
+
#FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FF,,F,,FF,F,,F,F,,,F,:,,,,:FFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FF

After fist commnad

@A00564:455:H3HKYDSX3:3:1101:1380:1000 1:N:0:ACTTGA
NTATCGCCCACGTGGACCATGGCAAGTCCACGCTGACGTTGCTAGAC
+
#FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
 
 The second command demultiplex the raeds based on 5-n adapter at the 3 end of the reads. I noticed that for the second command there is some leaking from one of my files (NI816) to other (NI813)  (NI813 is not what I expcet, there is more reads in this file than expceted) and the bases preceding removed adapters for NI816 barcode looks a bit odd with 0% of "C" shown below 

=== Adapter NI813 ===

Sequence: CTAGA; Type: regular 3'; Length: 5; Trimmed: 37008693 times

No. of allowed errors:
1-4 bp: 0; 5 bp: 1

Bases preceding removed adapters:
  A: 19.7%
  C: 23.4%
  G: 24.2%
  T: 32.7%
  none/other: 0.0%

=== Adapter NI816 ===

Sequence: TAGAC; Type: regular 3'; Length: 5; Trimmed: 87797594 times

No. of allowed errors:
1-4 bp: 0; 5 bp: 1

Bases preceding removed adapters:
  A: 28.6%
  C: 0.0%
  G: 30.2%
  T: 41.1%
  none/other: 0.0%

I think the issue is coming from NI813 barcode (C**TAGA**) and NI816 (**TAGA**C) which are similar but cannot find a way to solve it. These are pair end reads but I only use R1 to demultiplex and analyse my data.

Thanks for you help in advance and let me know if I can provide more info.

**Comment by mehdiamiri123** [comment]
It seems that anchoring the adapter to the read with $ is working ok, I will let you know if there is any other issue.

**Comment by marcelm** [comment]
You are absolutely right that the similarity between these barcodes leads to the problems. If anchoring is possible, you should do it.

It depends a bit on what you are sequencing, but the barcodes are so short that you could get random matches anywhere in the read just by chance. Since Cutadapt looks for the leftmost occurrence of the sequence if you don’t anchor it, that would also result in false matches.

**Comment by marcelm** [comment]
It appears I can close this issue now. Let me know if there are any open questions!
